assets/js/admin-settings/index.jsx

```diff
@@ -0,0 +1,86 @@
+/* global localizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect, useState } from 'react';
+import useSiteOptions from '../services/hooks/use-site-options';
+import Rule from './rule';
+
+
+const AdminSettings = () => {
+  const [{ loading, saving, settings }, setOptions] = useSiteOptions();
+
+  /**
+   * Helper function for saving the in-memory settings to the server.
+   */
+   const saveSettings = (rule, ruleIndex) => {
+    // Remove old rule.
+    let newVal = settings.apple_news_automation ?? [];
+    if (ruleIndex) {
+      newVal = settings.apple_news_automation.filter((x, index) => index !== ruleIndex);
+    }
+    // Add updated rule.
+    newVal.push(rule);
```

Comment:
The order of rules is going to be important, so we're going to need methods for being able to add a new one at the end, removing one at a particular position, editing one at a particular position, and re-ordering (likely remove at a particular position and then insert into a particular position). I have some prior art on this that I can share. I'd generally recommend handling the add/remove/edit/reorder functions separately from the save function.